pkg/controller/handler.go

```diff
@@ -204,56 +211,99 @@ func (c *Controller) generateVirtualService(ingress *networkingv1.Ingress, exist
 
 		// Add the path
 		for _, path := range rule.HTTP.Paths {
-
-			servicePort, err := c.getServicePort(ingress.Namespace, path.Backend)
+			routes, err := c.createHttpRoutesForPath(ingress, host, path, portsOnGateways)
 			if err != nil {
 				return nil, err
 			}
 
-			var authorityMatchType v1beta1.StringMatch
+			vs.Spec.Http = append(vs.Spec.Http, routes...)
+		}
+	}
 
-			if strings.Contains(host, "*") {
-				authorityMatchType = v1beta1.StringMatch{
-					MatchType: &v1beta1.StringMatch_Regex{
-						// Convert to Regex which is required by Envoy.
-						Regex: strings.ReplaceAll(strings.ReplaceAll(host, ".", "\\."), "*", ".*"),
-					},
-				}
-			} else {
-				authorityMatchType = v1beta1.StringMatch{
-					MatchType: &v1beta1.StringMatch_Exact{
-						Exact: host,
-					},
-				}
+	return vs, nil
+}
+
+// Returns the ports on the Gateway for Servers not running HTTPRedirect
+func (c *Controller) getNonHTTPPRedirectPortsOnGateways(gateways []*istionetworkingv1beta1.Gateway) []uint32 {
+	var ports []uint32
+
+	for _, gateway := range gateways {
+		for _, server := range gateway.Spec.Servers {
+			if !server.Tls.HttpsRedirect {
+				ports = append(ports, server.Port.Number)
 			}
+		}
+	}
 
-			route := &v1beta1.HTTPRoute{
-				Match: []*v1beta1.HTTPMatchRequest{
-					{
-						Authority: &authorityMatchType,
-					},
+	return ports
+}
+
+func (c *Controller) createHttpRoutesForPath(ingress *networkingv1.Ingress, host string, path networkingv1.HTTPIngressPath, portsOnGateways []uint32) ([]*v1beta1.HTTPRoute, error) {
```

Comment:
Mostly a nit pick.. but can we capitalize `HTTP` in the function name like in other functions?